File: growing-shop-infra/security/src/main/java/com/example/auth/LoginUser.java
package com.example.auth;

import com.example.domain.user.User;
import org.springframework.security.access.AccessDeniedException;
import org.springframework.security.core.context.SecurityContextHolder;

public class LoginUser {
    public static User getUserInSecurityContext() {
        Object principal = SecurityContextHolder.getContext().getAuthentication().getPrincipal();

        if (principal == null) {
            throw new AccessDeniedException("유저 정보가 없습니다.");
        }
        if (!(principal instanceof User)) {
            throw new AccessDeniedException("로그인 된 유저 정보의 데이터가 잘못되었습니다");
        }

        return (User) principal;
    }
}

File: growing-shop-api/auth/src/main/kotlin/com/example/growingshopauth/config/security/LoginUser.kt
package com.example.growingshopauth.config.security

import com.example.domain.user.User
import kotlinx.coroutines.reactive.awaitSingle
import org.springframework.security.core.context.ReactiveSecurityContextHolder

class LoginUser {
    companion object {
        suspend fun getUserInSecurityContext(): User {
            val authentication = ReactiveSecurityContextHolder.getContext().awaitSingle()
                ?: throw IllegalAccessException("로그인된 유저 정보가 없습니다")

            return authentication.authentication.principal as User
        }
    }
}
